ios/brave-ios/Sources/BraveShared/Extensions/URLExtensions.swift

```diff
@@ -233,6 +233,19 @@ extension URL {
     }
     return self
   }
+
+  // Returns true if a string is a valid URL, with the specified optional scheme,
+  // without percent encoding, idn encoding, or modifying the URL in any way.
+  public static func isValidURLWithoutEncoding(text: String, scheme: String? = nil) -> Bool {
+    if let components = URLComponents(string: text),
+      !(scheme ?? "").isEmpty ? components.scheme == scheme : true,
+      !(components.host ?? components.path).isEmpty
+    {
```

Comment:
This is a very confusing to read, better to split it up? Its supposed to check that the scheme matches if provided and that the either a host or path exist, right?

Reply:
Yeah correct. That's what it does exactly.

I'll split it up no worries.